nomos-core/chain-defs/src/block/mod.rs

```diff
@@ -59,3 +59,28 @@ impl<
         self.bl_blobs.len()
     }
 }
+
+impl<
```

Comment:
Now that I look at this, I think the `Block` struct does not need any bounds in its definition, since `IndexSet` does not require anything either. What do you think about removing those bounds from the `Block` definition, which would then imply remove all bounds but `DeserializeOwned` for the `Tx` type here? I think it would be much cleaner.

Reply:
If it's out of scope for this PR, I understand, although it's a pity to introduce all these bounds when they are not really needed, it seems.

Reply:
Unfortunately there is not. https://github.com/rust-lang/rust-clippy/issues/1463, https://users.rust-lang.org/t/unused-trait-bounds/37516. Reason being there is risk for high percentage of false positives, since in some cases a marker trait might be required. I have never myself come across such an instance (besides for `Send` and `Sync`, which could be special cases), and I would think an allow-by-default lint might be a good thing to have, but right now it does not exist.

Reply:
Yeah but the `Block` definition is a simple one, and removing trait bounds only from the struct definition is non breaking and relatively simple, in this specific case, that's why I suggested it. On a global scale, we probably have quite a bit of over-requirement of trait bounds, and I will try to pay particular attention to them when I review PRs, so we can remove them as we go, I'd say.